### PR TITLE
ITS: Implement AlpideResponseSimulation, on-the-fly root file generation, and more.

### PR DESCRIPTION
**Pull request by mconcas**
Dear @iouribelikov @AlexBigO,

TL;DR: proposed solution is to use a dedicated executable, that does the job of the previous macro, as you can see in the code.

Could you please check if this works for you, also for the digitisation, please?

Longer explanation:
 Macros are really painful to run during the installation, essentially because it's hard to satisfy all the recursive `INCLUDE_PATH`s of all the involved headers, not only the one you have finally in the macro.
There is no clean way to do so, and this makes also sense as multiple targets are being built in parallel in a non-predictable way (at least the uncorrelated ones).

As a personal comment, I would prefer to store all of these .txt files in a dedicated repository in AliceO2, and fetch them with the executable, so that we don't store 800+ more files in this code repository.
I'll let maybe comment @shahor02 and @ktf on this.

### CODE REVIEW

File: Detectors/ITSMFT/common/simulation/include/ITSMFTSimulation/DigiParams.h [hunk 1]
```diff
@@ -110,9 +110,9 @@ class DigiParams
   int mNSimSteps = 7;                      ///< number of steps in response simulation
   float mEnergyToNElectrons = 1. / 3.6e-9; // conversion of eloss to Nelectrons
 
-  float mVbb = 3.0;   ///< back bias absolute value for MFT (in Volt)
-  float mIBVbb = 3.0; ///< back bias absolute value for ITS Inner Barrel (in Volt)
-  float mOBVbb = 3.0; ///< back bias absolute value for ITS Outter Barrel (in Volt)
+  float mVbb = 0.0;   ///< back bias absolute value for MFT (in Volt)
+  float mIBVbb = 0.0; ///< back bias absolute value for ITS Inner Barrel (in Volt)
+  float mOBVbb = 0.0; ///< back bias absolute value for ITS Outter Barrel (in Volt)
```
Comment (shahor02) [inline Detectors/ITSMFT/common/simulation/include/ITSMFTSimulation/DigiParams.h:115]: perhaps one can get rid of duplication of all these parameters already defined in the DPLDigitizerParam  (legacy of O2 early days) and use this latter directly?

**Review by shahor02 — COMMENTED** [review]
Hi @mconcas 

Looks fine. For the dedicated repository for the response files, I agree that having it as a dependency of the O2 (or O2Sim) would be cleaner and would also facilitate having different versions of response files as different branches in this repo.

**Comment by mconcas** [comment]
> Hi @mconcas
> 
> Looks fine. For the dedicated repository for the response files, I agree that having it as a dependency of the O2 (or O2Sim) would be cleaner and would also facilitate having different versions of response files as different branches in this repo.

Then I'd create a dedicated repo for the response files. Maybe generic enough to host also the ones for ITS 3 (when they will be available), so to accommodate both.
Then with @AlexBigO we will restructure his contribution accordingly.

**Comment by mconcas** [comment]
This requires: https://github.com/alisw/alidist/pull/4701

File: Detectors/ITSMFT/common/data/AlpideResponseData/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+o2_add_executable(alpide-response-generator
+                  SOURCES AlpideResponse.cxx
+                  PUBLIC_LINK_LIBRARIES O2::ITSMFTSimulation
+                                        ROOT::Core
+                  TARGETVARNAME targetName)
+
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${CMAKE_SOURCE_DIR}/Detectors/ITSMFT/common/data/AlpideResponseData/AlpideResponse.cxx)
+
+add_custom_command(TARGET O2exe-alpide-response-generator POST_BUILD
+                   COMMAND ${CMAKE_BINARY_DIR}/stage/bin/o2-alpide-response-generator -i  $ENV{ITSRESPONSE_ROOT}/response/AlpideResponseData/ -o ${CMAKE_SOURCE_DIR}/Detectors/ITSMFT/common/data/AlpideResponseData/
```
Comment (iouribelikov) [inline Detectors/ITSMFT/common/data/AlpideResponseData/CMakeLists.txt]: The ROOT file with responses should be installed under $O2_ROOT/share.  (Not under $CMAKE_SOURCE_DIR)

**Review by iouribelikov — CHANGES_REQUESTED** [review]
Thank you very much, @mconcas !
The generation of ROOT file with responses works. 
But :
1) The file gets installed under the source directory, whereas the digitiser tries to get it from under the $O2_ROOT/share/.
2) Now, this PR needs to be rebased on dev.

File: Detectors/ITSMFT/common/data/AlpideResponseData/CMakeLists.txt (hunk 2 above)
Comment (mconcas) [reply Detectors/ITSMFT/common/data/AlpideResponseData/CMakeLists.txt]: > Thank you very much, @mconcas ! The generation of ROOT file with responses works. But :
> 
> 1. The file gets installed under the source directory, whereas the digitiser tries to get it from under the $O2_ROOT/share/.
> 2. Now, this PR needs to be rebased on dev.

Thanks, this I saw, and fixed in my local fork I was waiting to have the package in alidist.
I'll update the pr these days.

**Comment by mconcas** [comment]
@iouribelikov could you try with this?

**Review by iouribelikov — DISMISSED** [review]
For me, everything works now perfectly fine.
Thanks a lot, @mconcas.

**Comment by mconcas** [comment]
> For me, everything works now perfectly fine. Thanks a lot, @mconcas.

great, I fixed another hardcoded path in Digitizer.h, if you approve I can merge when/if it becomes green.

**Review by iouribelikov — DISMISSED** [review]
@mconcas  It looks like both of us are working on a Mac !  :-)

**Comment by mconcas** [comment]
> @mconcas It looks like both of us are working on a Mac ! :-)

True, but I have to tell that full_system_test script that runs in fullCI failed on mac too. So case insentitivity is not always that sloppy.
Hope this is last iteration on this... :)

**Comment by mconcas** [comment]
> @mconcas It looks like both of us are working on a Mac ! :-)

@iouribelikov, Sorry for dismissing review again, previos copy_file  command was creating issues with permission on file, making it impossible to delete it as normal user. Now should be fixed.

**Review by iouribelikov — APPROVED** [review]

**Comment by mconcas** [comment]
Hi @iouribelikov @shahor02, I think we miss either @sawenzel 's approval or a more powerful merger here.
I cannot merge without all required approvals.

**Comment by iouribelikov** [comment]
@mconcas @shahor02 @sawenzel  Yes. I think we could merge it.

**Comment by shahor02** [comment]
Hi, the fullCI has failed this morning because the ccdb server was down. 
Was the CI fully green before the power cut?

**Comment by mconcas** [comment]
Hi, yes it was. But I don't mind to wait another iteration.

**Comment by aphecetche** [comment]
@mconcas would it be possible to add a more explicit dependency on the new package `ITSResponse` at the cmake level on the O2 side ? Because as it is now the dependency is a bit hidden and only comes through alidist (and thus compilations not using alibuild only notice the dependency very late in the build...)

e.g. a `FindITSResponse.cmake` and the corresponding `find_package(ITSResponse)` ?

**Comment by mconcas** [comment]
> @mconcas would it be possible to add a more explicit dependency on the new package `ITSResponse` at the cmake level on the O2 side ? Because as it is now the dependency is a bit hidden and only comes through alidist (and thus compilations not using alibuild only notice the dependency very late in the build...)
> 
> e.g. a `FindITSResponse.cmake` and the corresponding `find_package(ITSResponse)` ?

@aphecetche: ok, I see. I'll try to set that up.
